[Revised] Update docs

bridadan opened the pull request:
This is an update of #214, but the `CONTRIBUTING` doc has been left in. We should create a new task to update this doc for the future.

@mazimkhan @AnotherButler Could you please take a look?

adbridge commented on README.md:
```diff
@@ -563,196 +96,87 @@ In the below example there are two builds defined:
 
-* We will first list the tests we want to execute:
+### Listing all tests
+We can use the `-l` arguement to list all availble tests:
```
Some typos - arguement and availble

adbridge commented on README.md:
```diff
@@ -563,196 +96,87 @@ In the below example there are two builds defined:
-        test 'mbed-drivers-test-c_strings'
-mbedgt: available tests for built 'K64F-GCC', location './.build/K64F/GCC_ARM'
-        test 'mbed-drivers-test-generic_tests'
+mbedgt: available tests for built 'K64F-ARM', location './BUILD/K64F/ARM'
```
shouldn't 'built' be 'build' ?

adbridge commented on README.md:
```diff
@@ -765,92 +189,39 @@ $ mbedls
 
-# Commissioning mbed platforms
+## Host test detection
+When developing with mbed OS, Grentea will detect host tests automatically if they placed in the correct location. All tests in mbed OS are placed under a `TESTS` directory. You may place custom host test scripts in a folder named `host_tests` in this folder. For more information on the mbed OS test directory structure, please see the [mbed CLI documentation](https://docs.mbed.com/docs/mbed-os-ha ...
```
"if they are placed in the correct location"

adbridge commented on CONTRIBUTING.md:
````diff
@@ -1,9 +1,203 @@
+```
+Now you can create separate branches in the forked repository and prepare pull requests with changes.
+### Issues solving
+Simple work-flow issue solving process may contain below steps:
````
A simple work-flow for solving issues may follow the steps below:

adbridge commented on CONTRIBUTING.md:
```diff
@@ -1,9 +1,203 @@
+* Create pull request from GitHub webpage (your fork's dashboard).
+
+### Branch naming conventions
+We prefer is you use standardised naming convention when creating pull requests.
```
We prefer you to use a standardised ...

adbridge commented on CONTRIBUTING.md:
```diff
@@ -1,9 +1,203 @@
+
+### Branch naming conventions
+We prefer is you use standardised naming convention when creating pull requests.
+Below few example of branch names' prefixes you could use when creating pull request from your fork:
```
Here are a few examples of branch name prefixes...

adbridge commented on CONTRIBUTING.md:
````diff
@@ -1,9 +1,203 @@
+* ```test_``` - when pull request will consist of only new/updates to test cases. E.g. ```test_paralllel_execution```.
+
+## Coding style and coding rules
+This chapter attempts to explain the basic styles and patterns that are used in mbed test tools projects. he following norms should be followed for new code, and for code that needs clean-up.
````
The following rules should be followed for producing new code and updating existing code.

adbridge commented on README.md:
```diff
@@ -94,15 +94,15 @@ In the below example there are two builds defined:
-## Command line usage
-Here we will highlight a few of the capabilities of the Greentea command line interface. For a full list of the availble options, please run `mbedgt --help`.
+## Command-line usage
+This section highlights a few of the capabilities of the Greentea command-line interface. For a full list of the availble options, please run `mbedgt --help`.
```
availble ?

bridadan commented:
@adbridge I've addressed your comments, however please note the CONTRIBUTING doc has other issues in it. For this reason I suggest we come back to that doc a later date.
